setup.py

```diff
@@ -69,6 +73,43 @@ def _load_py_module(name: str, location: str) -> ModuleType:
     return py
 
 
+def _named_temporary_file(directory: Optional[str] = None) -> str:
+    # `tempfile.NamedTemporaryFile` has issues in Windows
+    # https://github.com/deepchem/deepchem/issues/707#issuecomment-556002823
+    if directory is None:
+        directory = tempfile.gettempdir()
+    return os.path.join(directory, os.urandom(24).hex())
```

Comment:
You should be able to use `tempfile.mkdtemp(dir=directory)` directly.

Reply:
I was having windows issues: https://github.com/Lightning-AI/lightning/actions/runs/3398677421/jobs/5651928499#step:13:44

Reply:
probably because it needed to be an absolute path

Reply:
Unfortunately `setuptools.setup` requires a relative path